js/private/lifecycle/lifecycle-hooks.js

```diff
@@ -0,0 +1,78 @@
+const {
+    mkdirSync,
+    copyFileSync,
+    readdirSync,
+    readFileSync,
+    existsSync,
+    statSync,
+} = require('fs')
+const path = require('path')
+const { runPostinstallHooks } = require('@pnpm/lifecycle')
+
+async function main(argv) {
+    if (argv.length !== 2) {
+        console.error('Usage: node lifecycle-hooks.js [packageDir] [outputDir]')
+        process.exit(1)
+    }
+    const packageDir = argv[0]
+    const outputDir = argv[1]
+
+    copyPackageContents(packageDir, outputDir)
+
+    const packageJson = JSON.parse(
```

Comment:
This could be simplified to `const packageJson = require(path.join(packageDir, 'package.json'))`